test/file_test.cpp

```diff
@@ -6,10 +6,20 @@
 #include <iostream>
 #include <fstream>
 
+#include "tests_config.h"
 #include <wayfire/config/file.hpp>
 #include <wayfire/util/log.hpp>
 #include <wayfire/config/types.hpp>
 
+#if !HAS_GET_CURRENT_DIR_NAME
```

Comment:
Why not use getcwd unconditionally?

Reply:
yeah I guess we could just do `getcwd(NULL, _)` directly everywhere, it's not a posix standard thing either but both Linux and FreeBSD support that

…wait, is the cwd usage intentional? Why not just open relative paths?

Reply:
You do have a point here, relative paths could work just as fine.